packages/element/src/react-root.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * External dependencies
+ */
+import { createRoot, hydrateRoot } from 'react-dom/client';
+
+const mountedRoots = new WeakMap();
+
+/**
+ * Renders a given element into the target DOM node.
+ *
+ * @param {import('./react').WPElement} element Element to render.
+ * @param {HTMLElement}                 target  DOM node into which element should be rendered.
+ */
+export function render( element, target ) {
```

Comment:
As much as I'd like to just switch to the React 18 APIs and call it "done", I think we should play more responsibly towards the community here and keep exposing the legacy React 17 APIs. They might be necessary for consumers of the `@wordpress` packages that will upgrade to the React 18 dependency but will need additional time and effort to switch to the concurrent rendering mode. 

My personal opinion is that we could, and should be exposing those legacy APIs indefinitely, or at least until we upgrade to a React version that stops supporting them or completely drops them.

Reply:
`reinitializeEditor` is a public API though. There's a `wp.editPost.reinitializeEditor` function exposed that anyone can call.

Reply:
We could also deprecate it if we want to gracefully remove it.